src/wp-includes/navigation-areas.php

```diff
@@ -50,3 +50,202 @@ function get_navigation_areas() {
 	global $navigation_areas;
 	return $navigation_areas;
 }
+
+/**
+ * Migrates classic menus to a block-based navigation post on theme switch.
+ * Assigns the created navigation post to the corresponding navigation area.
+ *
+ * @since 5.9.0
+ * @access private
+ *
+ * @param string   $new_name  Name of the new theme.
+ * @param WP_Theme $new_theme New theme.
+ * @param WP_Theme $old_theme Old theme.
+ */
+function _migrate_menu_to_navigation_post( $new_name, $new_theme, $old_theme ) {
+	// Do nothing when switching to a theme that does not support site editor.
+	if ( ! wp_is_block_template_theme() ) {
+		return;
+	}
+
+	// get_nav_menu_locations() calls get_theme_mod() which depends on the stylesheet option.
+	// At the same time, switch_theme runs only after the stylesheet option was updated to $new_theme.
+	// To retrieve theme mods of the old theme, the getter is hooked to get_option( 'stylesheet' ) so that we
+	// get the old theme, which causes the get_nav_menu_locations to get the locations of the old theme.
+	$get_old_theme_stylesheet = static function() use ( $old_theme ) {
+		return $old_theme->get_stylesheet();
+	};
+	add_filter( 'option_stylesheet', $get_old_theme_stylesheet );
+
+	$locations    = get_nav_menu_locations();
+	$area_mapping = get_option( 'fse_navigation_areas', array() );
```

Comment:
Do we use the `fse_` prefix in core?

Reply:
+1 to this. Normal prefix with wp_

Reply:
+1

Reply:
https://github.com/WordPress/gutenberg/pull/36460